### PR TITLE
Do not release NSDateFormatter and instead reset property values if d…

### PR DESCRIPTION
**Pull request by irvingruan**
…ate formats change.

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @irvingruan, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by boxtcarpel** [comment]
We caught up offline. I think we should synchronize the stringFromDate: call in box_ISO8601String in NSDate+BOXAdditions instead. It's the overarching, simplest change that should cover all of the cases. Assuming that our diagnosis is correct and that code is the cause for the crash.

File: BoxContentSDK/BoxContentSDK/Categories/NSDate+BOXAdditions.m [hunk 1]
```diff
@@ -49,8 +49,10 @@ - (NSString *)box_ISO8601String
         dateFormatter.includeTime = YES;
         dateFormatter.defaultTimeZone = [[NSTimeZone alloc] initWithName:@"UTC"];
     });
-    
-    return [dateFormatter stringFromDate:self];
+
+    @synchronized (dateFormatter) {
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Categories/NSDate+BOXAdditions.m]: Lets be consistent with the method above.

File: BoxContentSDK/BoxContentSDK/Categories/NSDate+BOXAdditions.m [hunk 2]
```diff
@@ -49,8 +49,10 @@ - (NSString *)box_ISO8601String
...
+        return [dateFormatter stringFromDate:self];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Categories/NSDate+BOXAdditions.m]: lets set up the string here, inside the synchronized block, and then return it below.

That makes it easier to debug, for breakpoints etc

Comment (irvingruan) [inline BoxContentSDK/BoxContentSDK/Categories/NSDate+BOXAdditions.m]: +1

**Comment by boxtcarpel** [comment]
+2 LGTM

:diamond_shape_with_a_dot_inside:

**Pull request merged**